In [ ]:
from pathlib import Path
import matplotlib.pyplot as plt
import numpy as np
from PIL import Image

### Checking the dimensions and colour mode of a single ADNI image.

In [ ]:
img_path = Path("~/Datasets/ADNI/ADNI/AD_NC/train/NC/808819_88.jpeg").expanduser()


# load image into numpy array
with Image.open(img_path) as image:
    image.load()
    pixels = np.asarray(image)
    width, height = image.size

    # print image details
    print(f"Width: {width}px - Height: {height}px")
    print(f"Colour Mode: {image.mode}")
    print(f"Min Pixel Value: {pixels.min()}")
    print(f"Max Pixel Value: {pixels.max()}")



Width: 256px - Height: 240px
Colour Mode: L
Min Pixel Value: 0
Max Pixel Value: 255


## Checking that all images in dataset follow same format.

In [17]:
from collections import Counter

# get paths to all images in dataset
dataset_path = Path("~/Datasets/ADNI/ADNI/AD_NC").expanduser()
image_paths = [str(p) for p in dataset_path.rglob("*") if p.suffix.lower() == ".jpeg"]

image_info = []

# get details of every image
for path in image_paths:
    with Image.open(path) as image:
        image_info.append({"path": path, "mode": image.mode, "dimensions": image.size,})

print(f"Total images found: {len(image_paths)}")

Total images found: 30520


In [18]:
mode_counts = Counter(item["mode"] for item in image_info)
dimension_counts = Counter(item["dimensions"] for item in image_info)

print("\nColour modes:")
for mode, count in mode_counts.items():
    print(f"{mode}: {count} images")

print("\nDimensions:")
for dimensions, count in dimension_counts.items():
    print(f"{dimensions}: {count} images")


Colour modes:
L: 30520 images

Dimensions:
(256, 240): 30520 images


All images in dataset are grayscale and are (256 x 240) pixels.